### PR TITLE
BugFix: re-order the column as the json key

### PR DESCRIPTION
**Pull request by rickif**
It's faster to iterate the simdjson document as the key order of json.
ref: https://github.com/simdjson/simdjson/blob/master/doc/basics.md

This PR is to fix the mistake of the reordering.
ref: #1651 

# Benchmark Result:
test case: ndjson format, 95 MB per file, 100 files in total,
| Version | Throughput | 
| --- | --- | 
| fixed | 118MB/s | 
| main | 86MB/s |

### CODE REVIEW

**Review by chaoyli — DISMISSED** [review]

File: be/src/exec/vectorized/json_scanner.cpp [hunk 1]
```diff
@@ -463,13 +463,15 @@ void JsonReader::_reorder_column(std::vector<SlotDescriptor*>* slot_descs,
...
 
-        if (itr != slot_descs->end()) {
+        auto itr = std::find_if(ordered_slot_descs.begin(), ordered_slot_descs.end(),
+                                [&key](const SlotDescriptor* desc) { return desc->col_name() == key; });
+
+        if (itr != ordered_slot_descs.end()) {
             auto tmp = *itr;
-            *itr = slot_descs->at(idx);
-            slot_descs->at(idx) = tmp;
+            *itr = ordered_slot_descs.at(idx);
```
Comment (sduzh) [inline be/src/exec/vectorized/json_scanner.cpp:473]: Prefer operator `[]` than `at`

File: be/src/exec/vectorized/json_scanner.cpp [hunk 2]
```diff
@@ -463,13 +463,15 @@ void JsonReader::_reorder_column(std::vector<SlotDescriptor*>* slot_descs,
...
 
-        auto itr = std::find_if(slot_descs->begin(), slot_descs->end(),
-                                [key](const SlotDescriptor* desc) { return desc->col_name() == key; });
+        key_set.insert(key);
 
-        if (itr != slot_descs->end()) {
+        auto itr = std::find_if(ordered_slot_descs.begin(), ordered_slot_descs.end(),
+                                [&key](const SlotDescriptor* desc) { return desc->col_name() == key; });
+
+        if (itr != ordered_slot_descs.end()) {
```
Comment (sduzh) [inline be/src/exec/vectorized/json_scanner.cpp:471]: Are you trying to swap the element `*iter` with `ordered_slot_descs[idx]`? If so, using `std::swap(*iter, ordered_slot_descs[idx])`

Comment (rickif) [reply be/src/exec/vectorized/json_scanner.cpp:473]: Updated.

**Review by sduzh — DISMISSED** [review]

File: be/src/exec/vectorized/json_scanner.cpp [hunk 3]
```diff
@@ -463,13 +463,15 @@ void JsonReader::_reorder_column(std::vector<SlotDescriptor*>* slot_descs,
...
-        auto itr = std::find_if(slot_descs->begin(), slot_descs->end(),
-                                [key](const SlotDescriptor* desc) { return desc->col_name() == key; });
+        key_set.insert(key);
 
-        if (itr != slot_descs->end()) {
-            auto tmp = *itr;
-            *itr = slot_descs->at(idx);
-            slot_descs->at(idx) = tmp;
+        // Find the SlotDescriptor with the json document key.
+        auto itr = std::find_if(ordered_slot_descs.begin(), ordered_slot_descs.end(),
```
Comment (imay) [inline be/src/exec/vectorized/json_scanner.cpp:469]: Why not use a unordered_map<string, SlotDescriptor*> to accelerate this search?

**Review by chaoyli — DISMISSED** [review]

File: be/src/exec/vectorized/json_scanner.cpp (hunk 3 above)
Comment (rickif) [reply be/src/exec/vectorized/json_scanner.cpp]: @imay Updated.

**Review by imay — APPROVED** [review]
LGTM

**Review by sduzh — APPROVED** [review]

**Pull request merged**